lib/vanagon/component/source.rb

```diff
@@ -16,16 +16,21 @@ class << self
         # @param workdir [String] working directory to fetch the source into
         # @return [Vanagon::Component::Source] the correct subtype for the given source
         def source(uri, **options)
+          source_type = determine_source_type(uri)
```

Comment:
Seems like you probably shouldn't call `determine_source_type` if `source_type` was specified?